pkg/util/goroutinemap/goroutinemap.go

```diff
@@ -43,37 +43,42 @@ const (
 	maxDurationBeforeRetry time.Duration = 2 * time.Minute
 )
 
-// GoRoutineMap defines the supported set of operations.
-type GoRoutineMap interface {
-	// Run adds operationName to the list of running operations and spawns a new
-	// go routine to execute the operation. If an operation with the same name
-	// already exists, an error is returned. Once the operation is complete, the
-	// go routine is terminated and the operationName is removed from the list
-	// of executing operations allowing a new operation to be started with the
-	// same name without error.
-	Run(operationName string, operationFunc func() error) error
+// GoroutineMap defines the supported set of operations.
+type GoroutineMap interface {
+	// Run adds operationName/operationSubName to the list of running operations
+	// and spawns a new go routine to execute operationFunc.
+	// If an operation with the same operationName and no sub name already
+	// exists, an AlreadyExists or ExponentialBackoff error is returned.
+	// If an operation with the same operationName and operationSubName already
+	// exists, an AlreadyExists or ExponentialBackoff error is returned.
+	// Once the operation is complete, the go routine is terminated and the
+	// operationName/operationSubName is removed from the list of executing
+	// operations allowing a new operation to be started with the same name
+	// without error.
+	Run(operationName, operationSubName string, operationFunc func() error) error
```

Comment:
@saad-ali @smarterclayton The run interface was a nice and simple interface. We are burdending the compelxity of the "sub" logic into callers who don't care. We should add a new routine for the subLogic that allows for running on two dimensions as opposed to one. That also means callers don't have to use a "" second argument that is just confusing.

How about 
```
Run(operationName, operationSubName string, operationFunc func() error) error
RunSubTargets(target string,  targetSub string, operationFunc func() error) error
```